### PR TITLE
feat(Territorial): add presence

### LINKED ISSUE #6733: Presence for territorial.io
### Website name

territorial.io

### Website URL

https://territorial.io/

### Website logo

https://i.imgur.com/RLJLePH.png

### Prerequisites

- [ ] It is a paid service
- [ ] It displays NSFW content
- [ ] It is region restricted

### Description

Having a button to join the game (link to the home page or something similar).
It would also be cool to show some statistics on the game being played like singleplayer and multiplayer and maybe stuff like how much players are left, but i'm not sure if that's possible.

Description of game: Territorial.io is a strategy io game where players fight to expand their nations. Defend your territory, expand, and conquer the map in games of up to 500 players.

### PR DESCRIPTION
**Pull request by DarkVillager**
## Description 

Resolves #6733 
Added a presence for Territorial. 

The code structure uses js' way of pixel checking on a canvas. This is the only way to know what menu's are open etc. 
Sorry if this presence is low quality.
## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/T/Territorial/presence.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+enum Assets {
+	Logo = "https://i.imgur.com/8vXevly.png",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
```
Comment (EncryptedDev) [inline websites/T/Territorial/presence.ts:12]: `Assets.Logo` is only used once, you may as well just specify it here
```suggestion
			largeImageKey: "https://i.imgur.com/8vXevly.png",
```

File: websites/T/Territorial/presence.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			? (presenceData.details = "Viewing options")
+			: (tryOne = context.getImageData(500, 220, 400, 60).data[0]) === 14 ||
+			  tryOne === 6
+			? (presenceData.details = "Viewing color options")
+			: (tryOne = context.getImageData(500, 60, 400, 60).data[0]) === 14 ||
+			  tryOne === 6
+			? (presenceData.details = "Viewing color options")
+			: (presenceData.details = "Viewing main menu");
+
+		presenceData.buttons = [{ label: "View The Menu", url: href }];
```
Comment (EncryptedDev) [inline websites/T/Territorial/presence.ts:47]: What is The Menu?

Comment (DarkVillager) [reply websites/T/Territorial/presence.ts:47]: Depends?

File: websites/T/Territorial/presence.ts (hunk 1 above)
Comment (DarkVillager) [reply websites/T/Territorial/presence.ts:12]: okay

File: websites/T/Territorial/presence.ts (hunk 2 above)
Comment (EncryptedDev) [reply websites/T/Territorial/presence.ts:47]: Then the string doesn't make full sense

File: websites/T/Territorial/presence.ts (hunk 1 above)
Comment (redtrillix) [reply websites/T/Territorial/presence.ts:12]: not sure if this helps, but this is the favicon for the website: https://territorial.io/favicon.ico

File: websites/T/Territorial/presence.ts (hunk 2 above)
Comment (DarkVillager) [reply websites/T/Territorial/presence.ts:47]: The start menu/page is what I meant

Comment (EncryptedDev) [reply websites/T/Territorial/presence.ts:43]:
```suggestion
		presenceData.buttons = [{ label: "View Menu", url: href }];
```

File: websites/T/Territorial/presence.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+		presenceData.buttons = [{ label: "View Menu", url: href }];
+	} else if (
+		context.getImageData(1020, 540, 400, 60).data[0] === 255 ||
+		context.getImageData(1116, 502, 400, 60).data[0] === 255
+	)
+		presenceData.details = "The game has ended";
+	else {
+		presenceData.details = "Playing the game";
+		presenceData.buttons = [{ label: "Play The Game", url: href }];
```
Comment (EncryptedDev) [inline websites/T/Territorial/presence.ts:51]:
```suggestion
		presenceData.buttons = [{ label: "Play", url: href }];
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**